src/ext/ddtrace.c

```diff
@@ -0,0 +1,169 @@
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <Zend/zend.h>
+#include <Zend/zend_exceptions.h>
+#include <php.h>
+#include <php/ext/spl/spl_exceptions.h>
+#include <php_ini.h>
+#include <ext/standard/info.h>
+
+#include "compat_zend_string.h"
+#include "ddtrace.h"
+#include "dispatch.h"
+#include "dispatch_compat.h"
+
+#include "debug.h"
+
+#define UNUSED_1(x) (void)(x)
+#define UNUSED_2(x, y) \
+    do {               \
+        UNUSED_1(x);   \
+        UNUSED_1(y);   \
+    } while (0)
+#define UNUSED_3(x, y, z) \
+    do {                  \
+        UNUSED_1(x);      \
+        UNUSED_1(y);      \
+        UNUSED_1(z);      \
+    } while (0)
+#define _GET_UNUSED_MACRO_OF_ARITY(_1, _2, _3, ARITY, ...) UNUSED_##ARITY
+#define UNUSED(...) _GET_UNUSED_MACRO_OF_ARITY(__VA_ARGS__, 3, 2, 1)(__VA_ARGS__)
+
+#if PHP_VERSION_ID < 70000
+#define PHP5_UNUSED(...) UNUSED(__VA_ARGS__)
+#else
+#define PHP5_UNUSED(...) /* unused unused */
+#endif
+
+ZEND_DECLARE_MODULE_GLOBALS(ddtrace)
+
+#define ddtrace_disabled_guard()                                                               \
+    do {                                                                                       \
+        if (DDTRACE_G(disable)) {                                                              \
+            zend_throw_exception_ex(spl_ce_RuntimeException, 0 TSRMLS_CC,                      \
```

Comment:
What's the use case for disabling ddtrace where we want to interrupt execution with an exception? Or am I interpreting this incorrectly?

Reply:
It was meant to guard against accidentally using `dd_trace` and expecting it to work when its actually disabled. But after second thought if we silently NOOP this instead we would be able to avoid potential unwanted uncaught exceptions when someone simply disables the exception.

I'll change the implementation to not throw any exception and just do NOOP.